mobile/cafes_test.go

```diff
@@ -0,0 +1,241 @@
+}{
+	cafePath:    "./testdata/.textile3",
+	cafeApiPort: "5000",
+	mobilePath:  "./testdata/.textile4",
+}
+
+func TestMobile_SetupCafes(t *testing.T) {
+	var err error
+	cafesTestVars.mobile, err = createAndStartMobile(
+		cafesTestVars.mobilePath, false, &testHandler{}, &testMessenger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// start a cafe
+	_ = os.RemoveAll(cafesTestVars.cafePath)
+	err = core.InitRepo(core.InitConfig{
+		Account:     keypair.Random(),
+		RepoPath:    cafesTestVars.cafePath,
+		CafeApiAddr: "0.0.0.0:" + cafesTestVars.cafeApiPort,
+		CafeOpen:    true,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	cafesTestVars.cafe, err = core.NewTextile(core.RunConfig{
+		RepoPath: cafesTestVars.cafePath,
+		Debug:    true,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = cafesTestVars.cafe.Start()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	<-cafesTestVars.mobile.OnlineCh()
+	<-cafesTestVars.cafe.OnlineCh()
+}
+
+func TestMobile_RegisterCafe(t *testing.T) {
+	// create a token
+	token, err := cafesTestVars.cafe.CreateCafeToken("", true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// register with cafe
+	url := "http://127.0.0.1:" + cafesTestVars.cafeApiPort
+	err = cafesTestVars.mobile.RegisterCafe(url, token)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// add some data
+	err = addTestData(cafesTestVars.mobile)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	//flush := func() error {
+	//	count, err := flushCafeRequest(10)
+	//	if err != nil {
+	//		return err
+	//	}
+	//	if count > 0 {
+	//		return flush()
+	//	}
+	//}
+
+	_, err = flushCafeRequest(100)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+}
+
+func addTestData(m *Mobile) error {
+	thrd, err := addTestThread(m, &pb.AddThreadConfig{
+		Key:  ksuid.New().String(),
+		Name: "test",
+		Schema: &pb.AddThreadConfig_Schema{
+			Preset: pb.AddThreadConfig_Schema_MEDIA,
+		},
+		Type:    pb.Thread_PRIVATE,
+		Sharing: pb.Thread_INVITE_ONLY,
+	})
+	if err != nil {
+		return err
+	}
+
+	_, err = m.addFiles([]string{"../mill/testdata/image.jpeg"}, thrd.Id, "hi")
+	if err != nil {
+		return err
+	}
+
+	_, err = m.AddMessage(thrd.Id, "hi")
+	if err != nil {
+		return err
+	}
+
+	hash, err := m.addFiles([]string{"../mill/testdata/image.png"}, thrd.Id, "hi")
+	if err != nil {
+		return err
+	}
+
+	_, err = m.AddComment(hash.B58String(), "nice")
+	if err != nil {
+		return err
+	}
+
+	hash, err = m.addFiles([]string{"../mill/testdata/image.jpeg", "../mill/testdata/image.png"}, thrd.Id, "hi")
+	if err != nil {
+		return err
+	}
+
+	_, err = m.AddLike(hash.B58String())
+	if err != nil {
+		return err
+	}
+
+	_, err = m.AddMessage(thrd.Id, "bye")
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+/*
+Handle the request queue.
+  1. List some groups
+  2. List get the HTTP request list for each of those groups
+  3. Handle them (set to pending, send to cafe)
+  4. Delete failed (reties not handled here)
+  5. Set successful to complete
+*/
+func flushCafeRequest(limit int) (int, error) {
```

Comment:
^^ @asutula here's the process to implement in the SDK.  (fyi, this test method should read `flushCafeRequests` (plural)). The idea is that you would list the groups until there are zero.

Reply:
I'm going to remove that "group" terminology in the tests here cause it's not needed to understand the process.